pkg/context.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 Chair of Applied Cryptography, Technische Universität
+// Darmstadt, Germany. All rights reserved. This file is part of
+// perun-eth-demo. Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package pkg
+
+import (
+	"context"
+)
+
+// Context wraps a golang context/Context with a cancel function when available.
+// See https://golang.org/pkg/context/#Context
+type Context struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
+// Background returns the Background context that is never cancelled.
+// ref https://golang.org/pkg/context/#Background
+func Background() *Context {
+	return &Context{ctx: context.Background()}
+}
+
+// WithCancel returns a new Context that can cancelled with Context.Cancel() from the given Context.
+// ref https://golang.org/pkg/context/#WithCancel
+func (c *Context) WithCancel() *Context {
+	newCtx, cancel := context.WithCancel(c.ctx)
+	return &Context{ctx: newCtx, cancel: cancel}
+}
+
+// WithCancel returns a new Context that can cancelled with Context.Cancel() from the Background Context.
+// ref https://golang.org/pkg/context/#WithCancel
+func WithCancel() *Context {
+	return &Context{ctx: context.Background()}
+}
+
+// Cancel cancelles the context, only works on a Context created with Context.WithCancel().
+func (c *Context) Cancel() {
+	c.cancel()
```

Comment:
```suggestion
	if c.cancel != nil { c.cancel() }
```
so it also works with the background context and doesn't crash the whole application.